calculator.rb

```diff
@@ -0,0 +1,148 @@
+# I wanted to allow the user to enter their calculation in the same way you would in a normal calculator.
+# This ended up being much more of a headache than I originally anticipated,
+# particularly when dealing with negative numbers, hence this rather unweildy parsing method.
+# This method parses the user's input string and returns an array where the 1st element is the 1st number in the operation,
+# the 2nd element is the operator, and the 3rd element is the 2nd number in the operation.
+def parse(input)
+  # Removes white space, converts string into an array split by the possible operators (but retaining the operators).
+  # Removes empty strings from the array. I'm actually not sure where these are coming from, but I noticed them when the user enters a negative number.
+  parsed_input = input.gsub(/\s+/,"").split(/(\+)|(-)|(\*)|(\/)|(\^)|(%)|(add)|(subtract)|(multiply)|(divide)|(exponent)|(modulo)/).reject { |x| x.empty? }
+  # If the first element of the array is a negative sign,
+  # reassigns the 2nd element as a joined string of the sign and existing number in the 2nd element,
+  # and deletes the 1st element.
+  if parsed_input[0] == "-"
+    parsed_input[1] = parsed_input[0..1].join
+    parsed_input.slice!(0)
+  end
+  # If the 3rd element of the array is a negative sign,
+  # reassigns the 4th element as a joined string of the sign and existing number in the 4th element,
+  # and deletes the 3rd element.
+  if parsed_input[2] == "-"
+    parsed_input[3] = parsed_input[2..3].join
+    parsed_input.slice!(2)
+  end
+return parsed_input
+end
+
+# This method checks a few characteristics of the parsed array
+# including the length of the array (which should be = 3 after parsing),
+# and the format of the 1st and 3rd elements of the array, which should be numeric.
+def validation(parsed_array)
+  if parsed_array.length > 3
+    puts "Sorry, this calculator can't handle operations that complex."
+    # If the array length is greater than 3, the user likely entered more than one operator. Returns nil.
+  elsif parsed_array.length < 3
+    puts "Hmm. It doesn't look like the operation you've entered is valid for this calculator."
+    # If the array length is less than 3, the user likely didn't enter enough information,
+    # or didn't enter it in a manner that the parser could handle. Returns nil.
+  elsif (parsed_array[0] =~ /^[-+]?\d+(\.\d+)?$/).nil? || (parsed_array[2] =~ /^[-+]?\d+(\.\d+)?$/).nil?
```

Comment:
Nice regular expression, I would suggest to use `/^[-+]?\d*(\.\d+)?$/` instead of what you have there.  Yours requires at least 1 digit to the left of the decimal so .25 wouldn't work.